master/buildbot/process/build.py

```diff
@@ -325,7 +325,10 @@ def startBuild(self, build_status, workerforbuilder):
         # If it returns failure then we don't start a new build.
         if ready_or_failure is not True:
             yield self.buildPreparationFailure(ready_or_failure, "worker_prepare")
-            self.buildFinished(["worker", "not", "available"], RETRY)
```

Comment:
I dont think this is suffisiant. I would expect to have some code added to timeout the start of latent worker or something like this. Is that already there?